### PR TITLE
MySQLI tests

### PR DESCRIPTION
**Pull request by rjt-gupta**

### CODE REVIEW

File: tanner/emulators/mysqli.py [hunk 1]
```diff
@@ -6,7 +6,7 @@ def __init__(self, db_name):
         self.db_name = db_name
         self.helper = mysql_db_helper.MySQLDBHelper()
 
-    async def setup_db(self, query_map):
+    async def setup_db(self):
```
Comment (rnehra01) [inline tanner/emulators/mysqli.py]: Now you need to correct this where this method is called, so maybe do this in separate PR.

File: tanner/tests/test_mysqli.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+                await self.cursor.execute('CREATE DATABASE test_db')
+                await self.cursor.execute('USE {db_name}'.format(db_name='test_db'))
+                await self.cursor.execute('CREATE TABLE IF NOT EXISTS test (id INTEGER PRIMARY KEY, username TEXT)')
+                await self.cursor.execute('INSERT INTO test VALUES(0, "test0")')
+                await self.conn.commit()
+
+            self.loop.run_until_complete(connect())
+
+    @mock.patch('tanner.config.TannerConfig.get', side_effect=mock_config)
```
Comment (rnehra01) [inline tanner/tests/test_mysqli.py]: You can remove this.https://github.com/mushorg/tanner/pull/315/files#r287705380

File: tanner/tests/test_mysqli.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+        self.handler.helper.create_query_map = AsyncMock(
+            return_value={'comments': [{'name': 'comment_id', 'type': 'INTEGER'}, ],
+                          'users': [{'name': 'id', 'type': 'INTEGER'}, ]})
+
+        async def test():
+            self.returned_result = await self.handler.setup_db()
+
+        self.loop.run_until_complete(test())
+        self.handler.helper.create_query_map.assert_called_with(self.db_name)
```
Comment (rnehra01) [inline tanner/tests/test_mysqli.py]: What about this check?https://github.com/mushorg/tanner/blob/ab199c213bd58cba5f89a8ea6ac6a1d9b2a0b3aa/tanner/emulators/mysqli.py#L12

File: tanner/tests/test_mysqli.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        session = mock.Mock()
+        session.sess_uuid.hex = 'd877339ec415484987b279469167af3d'
+        attacker_db = 'attacker_' + session.sess_uuid.hex
+        self.handler.helper.copy_db = AsyncMock(return_value=attacker_db)
+
+        async def test():
+            self.attacker_db = await self.handler.create_attacker_db(session)
+
+        self.loop.run_until_complete(test())
+        self.handler.helper.copy_db.assert_called_with(self.db_name, attacker_db)
```
Comment (rnehra01) [inline tanner/tests/test_mysqli.py]: Check return value and this https://github.com/mushorg/tanner/blob/ab199c213bd58cba5f89a8ea6ac6a1d9b2a0b3aa/tanner/emulators/mysqli.py#L21

File: tanner/tests/test_mysqli.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    @mock.patch('tanner.config.TannerConfig.get', side_effect=mock_config)
+    def test_insert_dummy_data(self, m):
```
Comment (rnehra01) [inline tanner/tests/test_mysqli.py]: Test for this should be in `db_helper_test`?

File: tanner/tests/test_mysqli.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            await self.cursor.execute('SELECT * FROM test;')
+            self.returned_result = await self.cursor.fetchall()
+            await self.cursor.close()
+            self.conn.close()
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.returned_result, self.expected_result)
+
+    @mock.patch('tanner.config.TannerConfig.get', side_effect=mock_config)
+    def test_execute_query(self, m):
```
Comment (rnehra01) [inline tanner/tests/test_mysqli.py]: I think you should make multiple DBs, multiple tables in every DB, that will be better.

Comment (rnehra01) [reply tanner/tests/test_mysqli.py]: Also check for mysql query error.

File: tanner/emulators/mysqli.py (hunk 1 above)
Comment (rjt-gupta) [reply tanner/emulators/mysqli.py]: This is not called anywhere else apart from the tests.

File: tanner/tests/test_mysqli.py (hunk 3 above)
Comment (rjt-gupta) [reply tanner/tests/test_mysqli.py]: Done.

File: tanner/tests/test_mysqli.py (hunk 5 above)
Comment (rjt-gupta) [reply tanner/tests/test_mysqli.py]: Forgot to remove 😅.

File: tanner/tests/test_mysqli.py (hunk 3 above)
Comment (rnehra01) [reply tanner/tests/test_mysqli.py]: Check here NOT CALLED.

File: tanner/tests/test_mysqli.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
+import unittest
+import asyncio
+from unittest import mock
+from tanner.utils.asyncmock import AsyncMock
+from tanner.emulators.mysqli import MySQLIEmulator
+
+
+def mock_config(section, value):
```
Comment (rnehra01) [inline tanner/tests/test_mysqli.py]: config = {'host':'127.0.0.1' , 'user': 'root', 'passwod':''} and return config[value].
This would be better.

Comment (rjt-gupta) [reply tanner/tests/test_mysqli.py]: Coolz.

File: tanner/tests/test_mysqli.py [hunk 8]
```diff
@@ -4,14 +4,16 @@
 from tanner.utils.asyncmock import AsyncMock
 from tanner.emulators.mysqli import MySQLIEmulator
 
+config = {'host': '127.0.0.1', 'user': 'root', 'password': ''}
+
 
 def mock_config(section, value):
     if section == 'SQLI' and value == 'host':
```
Comment (rnehra01) [inline tanner/tests/test_mysqli.py]: These are not needed anymore.

Comment (rnehra01) [reply tanner/tests/test_mysqli.py]: Also put the dict inside the mock_config method.

Comment (rjt-gupta) [reply tanner/tests/test_mysqli.py]: Ohh Like that. Okayy

**Comment by rnehra01** [comment]
Looks good to me, I'll merge it once you resolve the -p flag issue in other PR.

File: tanner/tests/test_mysqli.py (hunk 7 above)
Comment (rnehra01) [reply tanner/tests/test_mysqli.py]: Do the same in other PR as well.

**Comment by rjt-gupta** [comment]
Updated the PR.

**Comment by rnehra01** [comment]
#315 merged.
